File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
import java.util.List;

class ResistorColorDuo {
    private final List<String> bandColors = List.of("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");
    
    int value(String[] colors) {
        return Integer.parseInt(
            String.format("%s%s", this.bandColors.indexOf(colors[0]), this.bandColors.indexOf(colors[1]))
        );
    }
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
COLOR_MAP = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9
}

def value(colors):
    return int(f"{COLOR_MAP.get(colors[0])}{COLOR_MAP.get(colors[1])}")
